diff --git a/src/Model/Group.js b/src/Model/Group.js
index 48d7471..5123444 100644
--- a/src/Model/Group.js
+++ b/src/Model/Group.js
@@ -3,48 +3,56 @@ import firebase from 'firebase'
 export function createGroup({
     name = 'DEFAULTNAME',
     memberCount = 0,
-    memberList = [],
-    videoList = []
-}) {
-    console.log("Group.createGroup")
+    members = {},
+    videos}) {
+    let log = "Group.createGroup"
+    console.log(log)
+
     const user = firebase.auth().currentUser
     if (user) {
         memberCount += 1
-        memberList.push(user.uid)
+        members[user.uid] = true
+
 
-        let ref = firebase.database().ref('groups/')
-        let groupRef = ref.push();
-        groupRef.set({
+        let headersRef = firebase.database().ref('groups/headers/').push();
+        let groupId = headersRef.key
+        
+        let data = {};
+        data['groups/headers/' + groupId] = {
             name: name,
             memberCount: memberCount,
-            memberList: memberList,
-            videoList: videoList,
-            owner: user.uid,
-        })
-        
-        console.log('SUCCESS: groupRef: ' + groupRef)
+            owner: user.uid
+        }
+        data['groups/members/' + groupId] = members
+        data['members/' + user.uid + '/groups/' + groupId] = true
+
+        return firebase.database().ref().update(data)
+            .then(() => console.log('SUCCESS: ' + log))
+            .catch(err => console.log('ERROR: ' + log + err))
     } else {
-        console.log("FAILED: couldn't get firebase user")
+        console.log("NO USER: couldn't get firebase user")
     }
 }
 
-export function getGroupById(id) {
+export function getGroupMembersById(groupId) {
     console.log("Group.getGroup")
-    if (user) {
-        memberCount += 1
-        memberList.push(user.uid)
-
-        firebase.database().ref('groups/')
-        .push({
-        name: name,
-        memberCount: memberCount,
-        memberList: memberList,
-        videoList: videoList,
-        owner: user.uid,
+
+    return firebase.database().ref('groups/members' + groupId).once('value')
+        .then(members => {
+            if (!members.exists()) {
+                return []
+            }
+            return Object.keys(members.val())
         })
-    } else {
-        console.log("FAILED: couldn't get firebase user")
-    }
+        .catch(err => console.log("FAILED: couldn't get firebase user"))
+}
+
+export function getGroupHeadersListener(onValue) {
+    console.log("Group.getGroupHeaders")
+    
+    return firebase.database().ref('groups/headers')
+        .on('child_added', onValue)
+
 }
 
 console.log('loaded Group.js')
\ No newline at end of file
